File: cmManagementSystem/src/main/java/com/example/cmManagementSystem/service/SponsorService.java
package com.example.cmManagementSystem.service;

import com.example.cmManagementSystem.dto.SponsorDto;
import com.example.cmManagementSystem.entity.Sponsor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface SponsorService {
    
    /**
     * Tüm sponsorları sayfalanmış şekilde getirir
     * 
     * @param pageable Sayfalandırma bilgileri
     * @return Sponsor sayfası
     */
    Page<SponsorDto> findAll(Pageable pageable);
    
    /**
     * Tüm sponsorları liste olarak getirir (SponsorController için)
     * 
     * @return Sponsor listesi
     */
    List<SponsorDto> findAll();
    
    /**
     * ID'ye göre sponsor getirir
     * 
     * @param id Sponsor ID
     * @return Sponsor
     */
    SponsorDto findById(Long id);
    
    /**
     * İsme göre sponsorları getirir
     * 
     * @param name Sponsor adı
     * @return Sponsor listesi
     */
    List<SponsorDto> findByName(String name);
    
    /**
     * Tipe göre sponsorları getirir
     * 
     * @param type Sponsor tipi
     * @return Sponsor listesi
     */
    List<SponsorDto> findByType(Sponsor.SponsorType type);
    
    /**
     * Seviyeye göre sponsorları getirir
     * 
     * @param level Sponsor seviyesi
     * @return Sponsor listesi
     */
    List<SponsorDto> findByLevel(Sponsor.SponsorLevel level);
    
    /**
     * Kulübe göre sponsorları getirir
     * 
     * @param clubId Kulüp ID
     * @return Sponsor listesi
     */
    List<SponsorDto> findByClubId(Long clubId);
    
    /**
     * Kulübe göre sponsorları getirir (SponsorController için)
     * 
     * @param clubId Kulüp ID
     * @return Sponsor listesi
     */
    List<SponsorDto> findByClub(Long clubId);
    
    /**
     * Etkinliğe göre sponsorları getirir
     * 
     * @param eventId Etkinlik ID
     * @return Sponsor listesi
     */
    List<SponsorDto> findByEventId(Long eventId);
    
    /**
     * Tipe ve seviyeye göre sponsorları sayfalanmış şekilde getirir
     * 
     * @param type Sponsor tipi
     * @param level Sponsor seviyesi
     * @param pageable Sayfalandırma bilgileri
     * @return Sponsor sayfası
     */
    Page<SponsorDto> findByTypeAndLevel(Sponsor.SponsorType type, Sponsor.SponsorLevel level, Pageable pageable);
    
    /**
     * Yeni sponsor oluşturur
     * 
     * @param sponsorDto Sponsor bilgileri
     * @return Oluşturulan sponsor
     */
    SponsorDto createSponsor(SponsorDto sponsorDto);
    
    /**
     * Sponsor bilgilerini günceller
     * 
     * @param id Sponsor ID
     * @param sponsorDto Güncellenecek bilgiler
     * @return Güncellenen sponsor
     */
    SponsorDto updateSponsor(Long id, SponsorDto sponsorDto);
    
    /**
     * Sponsoru siler
     * 
     * @param id Silinecek sponsor ID
     */
    void deleteSponsor(Long id);
    
    /**
     * Sponsoru bir kulübe ekler
     * 
     * @param sponsorId Sponsor ID
     * @param clubId Kulüp ID
     * @return Güncellenen sponsor
     */
    SponsorDto addSponsorToClub(Long sponsorId, Long clubId);
    
    /**
     * Yeni bir sponsor oluşturur ve bir kulübe ekler
     * 
     * @param clubId Kulüp ID
     * @param sponsorDto Sponsor bilgileri
     * @return Oluşturulan sponsor
     */
    SponsorDto addSponsorToClub(Long clubId, SponsorDto sponsorDto);
    
    /**
     * Sponsoru bir kulüpten çıkarır
     * 
     * @param sponsorId Sponsor ID
     * @param clubId Kulüp ID
     */
    void removeSponsorFromClub(Long sponsorId, Long clubId);
    
    /**
     * Sponsoru bir etkinliğe ekler
     * 
     * @param sponsorId Sponsor ID
     * @param eventId Etkinlik ID
     * @return Güncellenen sponsor
     */
    SponsorDto addSponsorToEvent(Long sponsorId, Long eventId);
    
    /**
     * Sponsoru bir etkinlikten çıkarır
     * 
     * @param sponsorId Sponsor ID
     * @param eventId Etkinlik ID
     */
    void removeSponsorFromEvent(Long sponsorId, Long eventId);
    
    /**
     * Kullanıcının belirtilen ID'li sponsoru yönetme yetkisi olup olmadığını kontrol eder
     * 
     * @param userId Kullanıcı ID
     * @param sponsorId Sponsor ID
     * @return Kullanıcı bu sponsoru yönetebilir mi
     */
    boolean isSponsorOfClub(Long userId, Long sponsorId);
} 

File: src/services/sponsorService.js
import { api } from './api';

/**
 * Sponsor ile ilgili işlemleri yöneten servis
 */
const sponsorService = {
  /**
   * Tüm sponsorları getirir
   * @param {object} params - Sayfalama ve filtreleme parametreleri
   * @returns {Promise} - Sponsorlar
   */
  getAllSponsors: (params = {}) => {
    return api.get('/sponsors', { params })
      .then(response => response.data);
  },
  
  /**
   * ID'ye göre sponsor getirir
   * @param {number} id - Sponsor ID
   * @returns {Promise} - Sponsor bilgileri
   */
  getSponsorById: (id) => {
    return api.get(`/sponsors/${id}`)
      .then(response => response.data);
  },
  
  /**
   * İsme göre sponsorları arar
   * @param {string} name - Sponsor adı
   * @returns {Promise} - Sponsorlar
   */
  getSponsorsByName: (name) => {
    return api.get(`/sponsors/name/${name}`)
      .then(response => response.data);
  },
  
  /**
   * Sponsor türüne göre sponsorları filtreler
   * @param {string} type - Sponsor türü
   * @returns {Promise} - Sponsorlar
   */
  getSponsorsByType: (type) => {
    return api.get(`/sponsors/type/${type}`)
      .then(response => response.data);
  },
  
  /**
   * Sponsor seviyesine göre sponsorları filtreler
   * @param {string} level - Sponsor seviyesi
   * @returns {Promise} - Sponsorlar
   */
  getSponsorsByLevel: (level) => {
    return api.get(`/sponsors/level/${level}`)
      .then(response => response.data);
  },
  
  /**
   * Kulübe göre sponsorları getirir
   * @param {number} clubId - Kulüp ID
   * @returns {Promise} - Sponsorlar
   */
  getSponsorsByClub: (clubId) => {
    return api.get(`/sponsors/club/${clubId}`)
      .then(response => response.data);
  },
  
  /**
   * Yeni sponsor oluşturur
   * @param {object} sponsorData - Sponsor bilgileri
   * @returns {Promise} - Oluşturulan sponsor
   */
  createSponsor: (sponsorData) => {
    return api.post('/sponsors', sponsorData)
      .then(response => response.data);
  },
  
  /**
   * Sponsor bilgilerini günceller
   * @param {number} id - Sponsor ID
   * @param {object} sponsorData - Güncellenecek sponsor bilgileri
   * @returns {Promise} - Güncellenen sponsor
   */
  updateSponsor: (id, sponsorData) => {
    return api.put(`/sponsors/${id}`, sponsorData)
      .then(response => response.data);
  },
  
  /**
   * Sponsor siler
   * @param {number} id - Sponsor ID
   * @returns {Promise} - İşlem sonucu
   */
  deleteSponsor: (id) => {
    return api.delete(`/sponsors/${id}`)
      .then(response => response.data);
  },
  
  /**
   * Sponsoru kulübe ekler
   * @param {number} sponsorId - Sponsor ID
   * @param {number} clubId - Kulüp ID
   * @returns {Promise} - İşlem sonucu
   */
  addSponsorToClub: (sponsorId, clubId) => {
    return api.post(`/sponsors/${sponsorId}/clubs/${clubId}`)
      .then(response => response.data);
  },
  
  /**
   * Sponsoru kulüpten kaldırır
   * @param {number} sponsorId - Sponsor ID
   * @param {number} clubId - Kulüp ID
   * @returns {Promise} - İşlem sonucu
   */
  removeSponsorFromClub: (sponsorId, clubId) => {
    return api.delete(`/sponsors/${sponsorId}/clubs/${clubId}`)
      .then(response => response.data);
  },
  
  /**
   * Sponsoru etkinliğe ekler
   * @param {number} sponsorId - Sponsor ID
   * @param {number} eventId - Etkinlik ID
   * @returns {Promise} - İşlem sonucu
   */
  addSponsorToEvent: (sponsorId, eventId) => {
    return api.post(`/sponsors/${sponsorId}/events/${eventId}`)
      .then(response => response.data);
  },
  
  /**
   * Sponsoru etkinlikten kaldırır
   * @param {number} sponsorId - Sponsor ID
   * @param {number} eventId - Etkinlik ID
   * @returns {Promise} - İşlem sonucu
   */
  removeSponsorFromEvent: (sponsorId, eventId) => {
    return api.delete(`/sponsors/${sponsorId}/events/${eventId}`)
      .then(response => response.data);
  }
};

export default sponsorService; 